airbyte-scheduler/src/main/java/io/airbyte/scheduler/persistence/DefaultJobPersistence.java

```diff
@@ -526,9 +539,29 @@ private void importTable(DSLContext ctx, final String schema, final DatabaseSche
         // LOGGER.debug(insertStep.toString());
         ctx.batch(insertStep).execute();
       }
+      final Optional<Field<?>> idColumn = columns.stream().filter(f -> f.getName().equals("id")).findFirst();
+      if (idColumn.isPresent())
+        resetIdentityColumn(ctx, schema, tableType);
+    }
+  }
+
+  private static void resetIdentityColumn(final DSLContext ctx, final String schema, final DatabaseSchema tableType) {
+    final Result<Record> result = ctx.fetch(String.format("SELECT MAX(id) FROM %s.%s", schema, tableType.name()));
+    final Optional<Integer> maxId = result.stream()
+        .map(r -> r.get(0, Integer.class))
+        .filter(Objects::nonNull)
+        .findFirst();
+    if (maxId.isPresent()) {
+      final Sequence<BigInteger> sequenceName = DSL.sequence(DSL.name(schema, String.format("%s_%s_seq", tableType.name().toLowerCase(), "id")));
+      ctx.alterSequenceIfExists(sequenceName).restartWith(maxId.get() + 101).execute();
     }
   }
 
+  private static void registerImportMetadata(final DSLContext ctx, final String airbyteVersion) {
+    ctx.execute(String.format("INSERT INTO airbyte_metadata VALUES(CURRENT_TIMESTAMP(0) || '_import_db', '%s');", airbyteVersion));
+    ctx.execute(String.format("UPDATE airbyte_metadata SET value = '%s' WHERE key = 'airbyte_db_version';", airbyteVersion));
```

Comment:
why do we need this query? isn't this record inserted as part of the import? it should already exist and be in the correct state, right?

Reply:
This is the import that is putting that record in?

or do you mean that we expect the migration to add such a record in the `airbyte_metadata` yaml file to be imported afterward with the correct value?

Reply:
good question. i think we expect the migration to update the existing value to the correct record. i don't think we expect the db have to do it itself.

(it's a good point though since the migratin script doesn't really do this right now :D)

Reply:
The import feature verifies that the VERSION file from the archive is good and then imports the data, it could just override and make sure the version in the DB is the correct one